Feature additional storm urls

clindeman opened the pull request.

clindeman committed:
feat: new virtual env for hhit only

Added a trimmed down version of hazpy that includes connection string file and changed username, this allows HHIT to be updated without impacting other tools like FAST and EXPORT, however it is decoupled from the hazpy package on anaconda.org but uses code copied from the github hazpy repo; removed hazpy install in virtual environment; new virtual environment just for HHIT; disabled hazpy update check as its no longer in this specific virtual environment; set virtual env to use python 3.9 from 3.7; added pdfrw library.

clindeman committed:
fix: condaChannel adjustment

clindeman committed:
fix: removed whitespace and updated openssl version from i to k

clindeman committed:
refactor: added back the hazpy dependency and hazpy checkfor updates

I found that disabling the "manage.checkForHazPyUpdates()" in update.py does not work since in manage.py under "def startApp(self, app_path, update_path):" as the 'res' variable is 0 whether the virtual environment from the config.json exists or not, so it skips the "if res != 0" that would have installed hazpy.

And the manage.py appears to rely on the "def checkForHazPyUpdates(self):" to install hazpy whether the virtual env existed or not.

I'm trying it out on the fema machine with the hazpy dependency in the environment.yaml and the update.py retaining the checkforhazpy updates. I suppose FHIT could still have hazpy in its environment and it could be updated as hazpy is updated.

clindeman committed:
fix: added print statement to replace 'pass' in try, except

clindeman committed:
refactor: reverted change to create separate virtual env for hhit only

HHIT will use hazus_env

clindeman committed:
feat: added check for auth zipfile download and safe_unzip function

The check for the auth zipfile is unique to HHIT and needs to line up with the config.json zipfile url. The safe_unzip function will ensure that the files being unzipped are not being unzipped to other directories but only into the current working directory.

clindeman committed:
fix: moved safe_unzip out of updateTool and added self variable

clindeman committed:
refactor: reverted/removed safe_unzip function and use

clindeman committed:
feat: Added Hazus icon; Renamed Hurrvac to Hurricane and added Hurrevac subtitle

clindeman committed:
feat: add safe_unzip function and replaced extractall with it

clindeman committed:
refactor: rename starting script file

hazus-hurrevac-import-tool.py was the previous name

clindeman committed:
chore: remove commented code

clindeman committed:
fix: changed name (#12)

Changed name from Hazus Hurrevac Import Tool to Hurricane Hazard Import Tool

clindeman committed:
Merge branch 'dev' into newdev

clindeman committed:
Squashed commit of the following:

commit 5444bc1ce563b18be2656a95c2510365204efe1d
Author: clindeman <clindeman@niyamit.com>
Date:   Mon Jun 21 15:35:23 2021 -1000

    fix: updated requirements

commit 59b40c34ec0fb0fe0cb7fab07ae9df7ddc5ee5d1
Author: clindeman <clindeman@niyamit.com>
Date:   Wed Jun 16 12:14:44 2021 -1000

    fix: changed formatting

commit 66fa9c38e2a2280b3751a679addaee03981d47f3
Author: clindeman <clindeman@niyamit.com>
Date:   Wed Jun 16 11:34:52 2021 -1000

    fix: cleanup of content and change to markdown

commit aaceacf126b08ff10cdaa9d7771bcdc93aac3ce8
Author: clindeman <clindeman@niyamit.com>
Date:   Wed Jun 16 10:33:04 2021 -1000

    fix: added missing step

commit fa906d2bd166c8607c48b76e4c4f4bee500fc8ee
Author: clindeman <clindeman@niyamit.com>
Date:   Wed Jun 16 10:19:02 2021 -1000

    fix: added troubleshooting section

commit ecd88adeb4cae623279d56d58d882344111a7eee
Author: clindeman <clindeman@niyamit.com>
Date:   Wed Jun 16 09:59:22 2021 -1000

    fix: added instructions on what to do if updating finds many conflicts. Renamed HITT to HHIT

commit a99bb9ddbf1e006bbd2f5be538e7114d53e733b0
Author: clindeman <clindeman@niyamit.com>
Date:   Tue Jun 15 12:10:16 2021 -1000

    fix: changed name

    Changed name from Hazus Hurrevac Import Tool to Hurricane Hazard Import Tool

clindeman committed:
Squashed commit of the following:

commit 5f7a75ef6c320a8bd0bfaedfe7ec723754749c3c
Author: clindeman <clindeman@niyamit.com>
Date:   Tue Jun 22 08:50:37 2021 -1000

    refactor: Check whoami to set proxy instead of waiting for timeout

    Also removed createProxyEnv method as it looks like it was not used.

commit 6ee076ca433f5452a642a0fca606a45f9ecddc8d
Author: Lindeman <0107902465@FEMA.DHS.GOV>
Date:   Mon Jun 21 19:16:01 2021 -1000

    fix: refactor code to work

    handleprody() now occurs before trying to access the url.

commit 1493b368fae7859db14e11643555ad8986733ae2
Author: clindeman <clindeman@niyamit.com>
Date:   Mon Jun 21 16:45:35 2021 -1000

    feat-Added Manage class to handle proxy needs

clindeman committed:
Merge branch 'dev' into newdev

clindeman committed:
fix: updated screenshots of release download

fix: updated readme with two new screenshots of downloading hhit.zip from the release page.

clindeman committed:
Add GPL3 license

Added License file to match other NHRAP github repositories' license.

clindeman committed:
Merge branch 'dev' into newdev

clindeman committed:
fix: add url for simulated storms

Add url in config for simulated storms. Add code to check both urls for storm data to support simulated storms as they are at a different url than the other storms. Add timeouts to all url requests to prevent infinite waiting. Change incorrect calls of getcode() to status_code.

clindeman committed:
feat: add url config entry for each storm type

Add a url config entry for each of the four storm types: active, historical, exercise, simulated so that they can be changed if needed.

clindeman committed:
Merge branch 'dev' into feature-additional-storm-urls

clindeman committed:
Merge branch 'dev' into feature-additional-storm-urls

clindeman committed:
Merge branch 'dev' into feature-additional-storm-urls

clindeman committed:
fix: increment tool version number

be-cunning commented on Python_env/hurrevac_storms.py:
```diff
@@ -95,12 +95,14 @@ def GetStormsJSON(self):
         manage = Manage()
         manage.handleProxy()
-        openUrl = requests.get(hurrevacSettings['HurrevacStormsURL'])
+        openUrl = requests.get(hurrevacSettings['HurrevacStormsURL'], timeout=10)
```
Why only 10 seconds? Is that enough time?  Also, should it re-try a few times (2 or 3 attempts) before officially going forward? We could add a while loop to try so.

clindeman replied:
Ten seconds should be enough time as the data is very small. Added code to try three times with two seconds in between each before raising an error.

be-cunning commented on Python_env/hurrevac_storms.py:
```diff
@@ -257,27 +259,49 @@ def GetStormJSON(self, StormId):
-            print("Error receiving data: %s" % openUrl.getcode())
-##            logging.error("Error receiving data: %s" % openUrl.getcode())
+            print("Error receiving data: %s" % openUrl.status_code)
+##          logging.error("Error receiving data: %s" % openUrl.status_code)
```
Can we remove this commented-out code?

clindeman replied:
We left the logging in as we were going to revisit, however we have yet to revisit after some time. There was an issue with the logging not working with the environment or the tool.

be-cunning commented on Python_env/hurrevac_storms.py:
```diff
@@ -257,27 +259,49 @@ def GetStormJSON(self, StormId):
+            https://hvx.hurrevac.com/hvx-api/v1/sim/advisories/storm/{stormidhere} has simulated storms and
+            empty json for non-simulated storms.
         """
 ##        logging.debug("Running GetStormJSON")
```
minor nit: why are the commented-out lines still here? That could cause some confusion down the road, plus, it makes the code less clean. Can we remove the commented-out lines?

clindeman replied:
Yes, I've removed some commented-out lines.

be-cunning commented on Python_env/hurrevac_settings.json:
```diff
@@ -9,7 +9,10 @@
                     	"Exercise":0, 
                     	"Simulated":0},
 	"HurrevacStormsURL":"https://hvx.hurrevac.com/hvx-api/v1/advisories/storms", 
-	"HurrevacStormURL":"https://hvx.hurrevac.com/hvx-api/v1/advisories/storm",
```
could you just add the "/" at the end of each one of these? If so, then you wouldn't have to concatenate it later on.

clindeman replied:
Yes, this will make the code cleaner.

clindeman committed:
fix: add forward slash to storm url

Add forward slash to end of Storm url in properties; Remove forward slash concatenation in hurrevac_storms.py

clindeman committed:
chore: remove commented lines

clindeman committed:
fix: add 3 tries for storms json

Add three tries for storms json with two seconds in between tries. Also alphabetize imports.
